llvm/lib/Transforms/Yk/ShadowStack.cpp

```diff
@@ -147,6 +150,40 @@ class YkShadowStackImpl {
     return false;
   }
 
+  // Insert a shadowstack_ptr debug variable when compiled with '-g' so that in
+  // gdb we can know the offset into the shadow stack for each frame. This uses
+  // llvm's debug intrinsic so does not affect optimized code.
+  void insertShadowDbgInfo(Function &F, DataLayout &DL, GlobalValue *SSGlobal,
+                           size_t SFrameSize) {
+    DISubprogram *SP = F.getSubprogram();
+    if (!SP) {
+      return;
+    }
+    DIBasicType *DISSTy = DIB->createBasicType("shadowstack", SFrameSize * 8,
```

Comment:
What's the magic `8` here?

Reply:
`createBasicType` requires size in bits, `SFrameSize` is size in bytes.